-- phpMyAdmin SQL Dump
-- version 3.2.4
-- http://www.phpmyadmin.net
--
-- Host: localhost
-- Generation Time: Jun 27, 2014 at 12:10 PM
-- Server version: 5.0.92
-- PHP Version: 5.3.26

--
-- Documentations from TopFIND v2-130522
--
SET SQL_MODE="NO_AUTO_VALUE_ON_ZERO";


/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
/*!40101 SET NAMES utf8 */;

--
-- Database: `topfind_production_130522`
--

-- --------------------------------------------------------

--
-- Table structure for table `documentations`
--
-- Creation: May 22, 2013 at 01:04 PM
--

CREATE TABLE IF NOT EXISTS `documentations` (
  `id` int(11) NOT NULL auto_increment,
  `name` varchar(255) collate utf8_unicode_ci default NULL,
  `title` varchar(255) collate utf8_unicode_ci default NULL,
  `long` text collate utf8_unicode_ci,
  `short` text collate utf8_unicode_ci,
  `category` varchar(255) collate utf8_unicode_ci default NULL,
  `position` int(11) default '0',
  `show` tinyint(1) default '0',
  `created_at` datetime default NULL,
  `updated_at` datetime default NULL,
  PRIMARY KEY  (`id`)
) ENGINE=InnoDB  DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci AUTO_INCREMENT=58 ;

--
-- Dumping data for table `documentations`
--

INSERT INTO `documentations` (`id`, `name`, `title`, `long`, `short`, `category`, `position`, `show`, `created_at`, `updated_at`) VALUES
(1, 'p-info', '', 'This section provides basic information retrieved from UniProtKB and MEROPS. Protein names and species and, if applicable, protease classification and isoforms are listed. The UniProtKB curated annotation is presented alongside the amino acid sequence. For further background information, links to the appropriate entries at UniProtKB and MEROPS are provided.', 'This section provides basic information retrieved from UniProtKB and MEROPS. Protein names and species and, if applicable, protease classification and isoforms are listed. The UniProtKB curated annotation is presented alongside the amino acid sequence. For further background information, links to the appropriate entries at UniProtKB and MEROPS are provided.', 'Protein information', 1, 1, '2011-03-17 15:40:03', '2011-04-15 14:28:17'),
(2, 'p-domains', 'Domains & features', 'The linear organization of termini, cleavage sites and stable chains as well as features and domains along the primary protein sequence is graphically represented. Moving the mouse over an element will provide additional information such as exact position, type and description.', 'The linear organization of termini, cleavage sites and stable chains as well as features and domains along the primary protein sequence is graphically represented. Moving the mouse over an element will provide additional information such as exact position, type and description.', 'Protein information', 2, 1, '2011-03-17 15:41:01', '2011-03-17 15:41:01'),
(3, 'f-protein', 'Evidence filter', 'By default, all known data is incorporated into the results page. This can be customized through a filter accessible at the top right of the page. After filtering, only data that is backed with evidence matching the given filter settings is incorporated. This will affect the termini, cleavage sites and substrates listed. Also the network view will only consider matching connections and the cleavage site preference is only built on data matching the criteria. ', 'By default, all known data is incorporated into the results page. This can be customized through a filter accessible at the top right of the page. After filtering, only data that is backed with evidence matching the given filter settings is incorporated. This will affect the termini, cleavage sites and substrates listed. Also the network view will only consider matching connections and the cleavage site preference is only built on data matching the criteria. ', 'Filter', 1, 1, '2011-03-17 16:27:17', '2011-04-15 15:11:18'),
(4, 'p-network', 'Network neighborhood', 'Network view shows the interconnection with other proteins through cleavage, inhibition and protein protein interactions. Some proteins are combined for conciseness and clarity of the network.\r\nThe dynamic network representation is realized using <a href="http://cytoscapeweb.cytoscape.org/">Cytoscape Web</a>\r\n<br/>\r\nLegend: \r\n<ul>\r\n<li>filled red node: displayed protein</li>\r\n<li>V-shaped node: protease</li>\r\n<li>rhomboid node: protease inhibitor</li>\r\n<li>circular node: other protein</li>\r\n<li>blue arrows: cleavage</li>\r\n<li>red tee bar: inhibition of proteolytic activity</li>\r\n<li>grey line: other protein-protein interaction</li>\r\n</ul>\r\n<br/>\r\nActions:\r\n<ul>\r\n<li>click and hold on free space: move canvas</li>\r\n<li>click and hold on protein: drag node</li>\r\n<li>double click on protein: open protein page</li>\r\n</ul>', 'Network view shows the interconnection with other proteins through cleavage, inhibition and protein protein interactions', 'Protein information', 3, 1, '2011-03-17 17:00:01', '2011-04-16 07:01:09'),
(5, 'p-termini', 'Termini', 'N- and C-termini are listed according to their position. Evidence information is given in brief tabular format and a link to full details, publications and data repository is provided. They are put into perspective by listing features and domains such as propeptides whose start or endpoint overlaps with the terminus.', 'N- and C-termini are listed according to their position. Evidence information is given in brief tabular format and a link to full details, publications and data repository is provided. They are put into perspective by listing features and domains such as propeptides whose start or endpoint overlaps with the terminus.', 'Protein information', 4, 1, '2011-03-17 17:00:55', '2011-03-17 17:00:55'),
(6, 'p-processing', 'Proteolytic processing', 'For every processing site within the protein, all proteases known to cleave at this site are listed alongside a brief table indicating the evidence that led to this observation. Detailed information is available through a link (Fig S2c). \r\nIf applicable, features (such as mutations) or feature boundaries (such as the end of a signal peptide) that overlap with the cleavage site are listed and depicted as grey bars on a blue protein backbone along with a vertical red line indicating the site of cleavage. ', 'For every processing site within the protein, all proteases known to cleave at this site are listed alongside a brief table indicating the evidence that led to this observation. Detailed information is available through a link (Fig S2c). \r\nIf applicable, features (such as mutations) or feature boundaries (such as the end of a signal peptide) that overlap with the cleavage site are listed and depicted as grey bars on a blue protein backbone along with a vertical red line indicating the site of cleavage. ', 'Protein information', 5, 1, '2011-03-17 17:02:03', '2011-03-17 17:02:03'),
(7, 'p-substrates', 'Substrates', 'For each substrate a link to the respective entry as well as a list of cleavage sites and corresponding evidence in short tabular format is given.', 'For each substrate a link to the respective entry as well as a list of cleavage sites and corresponding evidence in short tabular format is given.', 'Protein information', 6, 1, '2011-03-17 17:03:18', '2011-03-17 17:03:18'),
(8, 'p-pref', 'Cleavage site specificity', 'If the protein is a protease the cleavage site preference is displayed. The sequence preference is calculated based on all protein and peptide substrates that match the current filtering criteria. The information is visualized in two forms.\r\n<ol>\r\n<li><strong>Heatmap:</strong> Representation of the cleavage site preference showing the relative abundance of each amino acid at positions P5 to P5'' on a color scale from blue (0%) to red (100%). No correction for the natural abundance of aminiacids is made. </li>\r\n<li><strong>iceLogo:</strong> Representation of amino acids showing statistically significant (p &gt; 0.5) over- or under-representation compared to their natural abundance. Generated by the <a href="http://iomics.ugent.be/icelogoserver/logo.html">iceLogo Server</a> at the University of Ghent.</li>\r\n</ol>', 'If the protein is a protease the cleavage site preference is displayed. The sequence preference is calculated based on all protein and peptide substrates that match the current filtering criteria. The information is visualized as heatmap and iceLogo. ', 'Protein information', 7, 1, '2011-03-17 17:07:42', '2011-04-15 21:02:37'),
(9, 's-modification', 'Terminus modification', 'To restrict the search to proteins containing a specific amino acid modification at their N- or C-terminus select the modification of interest from the modification drop-down.<br/>\r\nOnly modifications that have been identified by at least one terminus are listed. More information on each modification can be found in the list of terminal <a href="/topfind/terminusmodifications">modifications</a>.<br/>\r\nSource: Information on modifications is based on the post-translational-modifications list provided by UniProtKB.\r\n', 'Restrict the search to proteins containing a specific amino acid modification at their N- or C-terminus', 'Search', 3, 1, '2011-03-19 20:40:10', '2011-11-15 22:42:47'),
(10, 's-function', 'Search by function', 'To restrict the search to proteases or protease inhibitors select the respective option from the function drop-down.<br/>\r\nSource: All proteins with at least one report of cleavage of a protein or peptide in TopFIND are considered a protease. Respectively all proteins with at least one inhibitory activity reported are considered an inhibitor for this listing.', 'Restrict the search to proteases or inhibitors of proteases', 'Search', 4, 1, '2011-03-19 20:46:15', '2011-06-22 23:11:16'),
(11, 's-chromosome', 'Chromosome location', 'Chromosome identifier (e.g. ''6'' for human chromosome 6)', 'Chromosome identifier (e.g. ''6'' for human chromosome 6)', 'Search', 6, 1, '2011-03-19 20:58:07', '2011-10-06 03:34:27'),
(12, 's-main', 'Search', 'Proteins can be searched for by:\r\n<ul>\r\n<li>UniProt ID or accession</li>\r\n<li>Protein or gene name (full or part of)</li>\r\n<li>IPI, Genebank and other identifiers</li>\r\n<li>MEROPS protease family identifier</li>\r\n</ul> \r\n\r\nThe search for a MEROPS protease family will retrieve all family members. ', 'Search by:\r\n<ul>\r\n<li>UniProt ID or accession</li>\r\n<li>Protein or gene name (full or part of)</li>\r\n<li>IPI, Genebank and other identifiers</li>\r\n<li>MEROPS protease family identifier</li>\r\n</ul> ', 'Search', 2, 1, '2011-03-19 21:26:02', '2011-04-16 07:04:24'),
(13, 'p-annotation', 'Protein annotation', 'General protein annotation includes protein names, genomic location of the encoding gene, isoform information, amino acid sequence, post translational modification, biochemical activity role in disease and more.\r\n<br/>\r\nSource: UniProtKB', 'Derived in unmodified form from UniProtKB', 'Protein information', 1, 1, '2011-03-19 21:51:53', '2011-04-15 21:11:22'),
(14, 'e_directness', 'Directness of observation', 'Directness is a qualitative measure used by TopFIND to describe if the reported event (terminus, cleavage or inhibition) was observed directly (e.g. edman sequencing of a protein terminus.) or indirectly (e.g. indirect observation of a cleavage through direct assaying the occurrence of a new protein terminus).', 'Indicates if the assay monitors the reported cleavage or terminus directly.', 'Evidence', 2, 1, '2011-03-19 22:46:11', '2011-06-22 23:11:24'),
(15, 'e_physrel', 'Physiological relevance', 'A qualitative measure of the likelihood that the identified terminus, cleavage or inhibition is present and relevant <it>in vivo</it>. ', 'Qualitative measure of the likelihood that the identified terminus, cleavage or inhibition is present and relevant <it>in vivo</it>. ', 'Evidence', 3, 1, '2011-03-20 01:21:33', '2011-04-15 21:12:31'),
(16, 'e_tissue', 'Tissue distribution', 'For <i>in vivo</i> observations the investigated tissues are listed.', 'Indicates the tissues in which the observations have been made', 'Evidence', 4, 1, '2011-03-20 01:25:55', '2011-04-15 21:12:47'),
(17, 'e_source', 'Information source (electronically)', 'If the stated information is derived from an external database this information is given. Where possible a direct link to the respective web resource is provided.', 'The external database the data is derived from', 'Evidence', 5, 1, '2011-03-20 01:28:47', '2011-04-15 21:13:07'),
(18, 'e_method', 'Method (other)', 'Assay or methodology used to make the identification. Please use this field to specify the method used if the ''Method'' drop-down does not list your method. ', 'Assay or methodology used to make the identification. Please use this field to specify the method used if the ''Method'' drop-down does not list your method. ', 'Evidence', 3, 0, '2011-03-20 01:30:22', '2011-10-06 01:23:35'),
(19, 'e_lab', 'Information source (research laboratory)', 'The laboratory that performed the assays and reported the finding.', 'The laboratory that performed the assays and reported the finding', 'Evidence', 6, 1, '2011-03-20 01:37:35', '2011-04-15 21:13:23'),
(20, 'e_evidence', '', 'Information on termini, cleavages and inhibition is always accompanied by one or several evidences. An evidence contains all available data grouped into different categories (see below) that supports the information. <br/>\r\nEvidences can be used to limit the information given for a protein to those supported by a specific evidence (see filter section).', '', 'Evidence', 0, 0, '2011-03-20 01:48:39', '2011-04-15 21:14:55'),
(21, 'e_code', 'Type of evidence ', 'A categorization of evidences based on a standard controlled vocabulary is provided. This allows for easy cross referencing to and from information in external databases.<br/>\r\nSource: UniProt evidencecodes (<a href="http://www.obofoundry.org/cgi-bin/detail.cgi?id=evidence_code">http://www.obofoundry.org/cgi-bin/detail.cgi?id=evidence_code</a>)  ', 'Categorization according to the controlled evidencecode vocabulary', 'Evidence', 5, 1, '2011-03-20 01:53:36', '2011-04-15 21:15:33'),
(22, 'f-modification', 'Modification filter', 'When browsing <a href="/topfind/terminusmodifications">modifications</a> of protein termini the displayed modifications can be restricted by:<br/>\r\n<ol>\r\n<li>The type of modification. This groups a number of similar modifications like ''N2-acetylarginine'' and ''N-acetylalanine'' into the ''acetylation'' category. \r\n<li>The protein terminus that becomes modified.</li>\r\n</ol>', 'Restricts the displayed modifications to a specific categroy (e.g. acetylation) or their presence at a specific protein terminus', 'Filter', 3, 1, '2011-03-20 02:08:06', '2011-11-15 22:50:20'),
(23, 'e_name', 'Evidence name ', 'Every evidence has a unique name that can be used for referencing and filtering. For published work it is usually in the form: ''first author - start of publication title''<br/>\r\nOther forms are possible', 'The evidence name identifies a specific evidence entry. Used to limit to one ore more specific evidences', 'Evidence', 1, 1, '2011-03-20 02:12:33', '2012-07-14 02:33:41'),
(24, 'p-boundaries', 'Affected feature boundaries', 'TopFIND detects any positional overlap between a cleavage or terminus and start or end of protein features such as domains and mutations as derived from UniProtKB.<br/>\r\n ', 'Start or end of protein features like domains or mutations that fall together with the terminus or cleavage', 'Protein information', 6, 1, '2011-03-20 02:19:43', '2011-06-22 23:11:34'),
(25, 'p-features', 'Affected protein features', 'TopFIND detects any positional overlap between a cleavage or terminus and protein features such as domains and mutations as derived from UniProtKB.<br/>', 'The cleavage or terminus is located within the stated protein features', 'Protein information', 6, 1, '2011-03-20 02:21:25', '2011-06-22 23:11:41'),
(26, 'p-cytoscape', 'Network viewer', '<br/>\r\nLegend: \r\n<ul>\r\n<li>filled red node: displayed protein</li>\r\n<li>V shaped node: protease</li>\r\n<li>romboid node: protease inhibitor</li>\r\n<li>circular node: other protein</li>\r\n<li>blue arrows: cleavage</li>\r\n<li>red tee bar: inhibition of proteolytic activity</li>\r\n<li>grey line: other protein-protein interaction</li>\r\n</ul>\r\n<br/>\r\nActions:\r\n<ul>\r\n<li>click and hold on free space: move canvas</li>\r\n<li>click and hold on protein: drag node</li>\r\n<li>double click on protein: open protein page</li>\r\n</ul>', 'Legend:\r\n<ul>\r\n<li>filled red node: displayed protein</li>\r\n<li>V shaped node: protease</li>\r\n<li>romboid node: protease inhibitor</li>\r\n<li>circular node: other protein</li>\r\n<li>blue arrows: cleavage</li>\r\n<li>red tee bar: inhibition of proteolytic activity</li>\r\n<li>grey line: other protein-protein interaction</li>\r\n</ul>\r\n<br/>\r\nActions:\r\n<ul>\r\n<li>click and hold on free space: move canvas</li>\r\n<li>click and hold on protein: drag node</li>\r\n<li>double click on protein: open protein page</li>\r\n</ul>', 'Protein information', 4, 0, '2011-03-21 18:31:24', '2011-04-15 21:18:41'),
(27, 'ne_description', '', '', 'Please provide a brief description on how the data has been obtained', 'Create new evidence', 2, 0, NULL, '2011-04-15 21:18:52'),
(28, 'ne_evidencefile', '', '', 'To provide detailed or supplementary information (especially for unpublished data) a pdf file can be selected for upload and attached to the evidence information', 'Create new evidence', 3, 0, NULL, '2011-04-15 21:19:10'),
(31, 'e_repository', 'Raw data repository', 'To make the underlying raw data accessible a link to the raw data stored in an online repository like pride or tranche can be provided', 'Link to the raw data stored in an online repository (e.g. pride or tranche)', 'Evidence', 5, 0, NULL, '2011-04-15 21:19:39'),
(32, 'ne_publications', '', '', 'Please provide one or several Pubmed IDs for published data', 'Create new evidence', 4, 0, NULL, '2011-04-15 21:19:49'),
(33, 'i-name', '', '', 'Please provide a unique name to identify this dataset import', 'Import', 0, 0, NULL, '2011-04-15 21:20:19'),
(34, 'i-description', 'Description', '', 'Brief description of the submitted data to assist the validation of file and data integrity.', 'Import', 0, 0, NULL, '2011-04-15 21:20:31'),
(35, 'i-evidence', '', '', 'Please select the appropriate evidence (created in step I)', 'Import', 0, 0, NULL, '2011-04-15 21:20:38'),
(36, 'i-csvfile', '', '', 'Locate and select a comma separated (csv) or excel file containing the data in the described format', 'Import', 0, 0, NULL, '2011-04-15 21:20:48'),
(37, 's-familysubstrates', 'Merops family substrates', 'Entering a MEROPS protease family or clan ID into the search field retrieves all members of the family. A list of all proteins cleaved by members of this family or clan is also provided. This list however is not affected by the filter settings (e.g. species).', 'These proteins are cleaved by members of the MEROPS protease family or clan provided in the search. The list is not filtered.', 'Search', 7, 1, '2011-03-22 15:15:14', '2011-04-15 21:21:10'),
(38, 's-intro', '', 'TopFIND provides search and filter functionality based on protein information and supporting evidence. A general protein search can be made through the search field in the menu bar on every page as well as through the search and advanced search on the start page. Filter options are displayed in the right column where available.', '', 'Search', 1, 1, '2011-04-15 14:33:52', '2011-06-22 23:10:28'),
(39, 'c-intro', '', '<p>Your data submission his highly valuable to us. We have assembled a <a href="/topfind/downloads/TopFIND â€“ HowTo contribute your data.pdf" target="_blank">short pictured guide [pdf]</a> on how to enter your experimental information and batch upload you protein termini, cleavages, cleavage sites or inhibitions. Please contact us if there are any questions or you want to enquire about other options for batch data improt (e.g. when you already have your data in a structured forma, or for example in a database). Even though the 3 step process below is very straight forward we are always happy to find even more efficient ways.</p> \r\n<p>Please submit your latest termini and protease cleavage identifications along with experimental meta data.</p>\r\n\r\nExperimental meta data can be added through a web form with detailed help available.\r\n\r\nNew data sets can be batch-imported from comma separated (csv) files in the following format:\r\n<ol>\r\n					<li><strong>proteases cleaving a native protein</strong><br>\r\n						please provide the data in the following order using the headerline: <i>protease,position,substrate</i><br>\r\n						<ul>\r\n						<li><strong>protease:</strong>UniProt ID: <i>Q9BYF1</i></li>\r\n						<li><strong>position:</strong>cleavage site position in the form: <i>222-223</i></li>\r\n\r\n						<li><strong>substrate:</strong>UniProt ID: <i>Q9BYF1</i></li>\r\n						</ul>\r\n					</li>\r\n					<li><strong>proteases cleaving a peptide</strong><br>\r\n						please provide the data in the following order using the headerline: <i>protease,peptide</i><br>\r\n						<ul>\r\n\r\n						<li><strong>protease:</strong>UniProt ID: <i>Q9BYF1</i></li>\r\n						<li><strong>peptide:</strong>peptide sequence with the cleavage site indicated by a colon: <i>GA:GQCVFA</i></li>\r\n						</ul>\r\n					</li>\r\n					<li><strong>proteases inhibited by protein inhibitors</strong><br>\r\n\r\n						please provide the data in the following order using the headerline: <i>protease,inhibitor</i><br>\r\n						<ul>\r\n						<li><strong>protease:</strong>UniProt ID: <i>Q9BYF1</i></li>\r\n						<li><strong>peptide:</strong>UniProt ID: <i>Q9BYF1</i></li>\r\n						</ul>\r\n\r\n					</li>\r\n					<li><strong>protein N-termini</strong><br>\r\n						please provide the data in the following order using the headerline: <i>protein,position,modification,confidence,confidence_type</i><br>\r\n						<ul>\r\n						<li><strong>protein:</strong>UniProt ID: <i>Q9BYF1</i></li>\r\n						<li><strong>position:</strong>first amino acid relative to uniprot full length protein sequence: <i>222</i></li>\r\n\r\n						<li><strong>modification:</strong>name of the modification of the first amino acid according to TopFIND <a href="/modifications">modifications</a>: <i>acteylation</i> or <i>N-acetylaspartate</i> </li>\r\n\r\n<li><strong>confidence: </strong>a confidence measurement if applicable (e.g. when based on peptide identification by mass spectrometry): <i>0.95</i></li>\r\n						<li><strong>confidence type: </strong>The type of the confidence value given (any of: ''unknown'',''probability'',''MASCOT score'',''X! Tandem score'',''PeptideProphet probability''): <i>probability</i></li>\r\n						</ul>\r\n					</li>\r\n					<li><strong>protein C-termini</strong><br>\r\n\r\n						please provide the data in the following order using the headerline: <i>protein,position,modification,confidence,confidence_type</i><br>\r\n						<ul>\r\n						<li><strong>protein:</strong>UniProt ID: <i>Q9BYF1</i></li>\r\n						<li><strong>position:</strong>last amino acid relative to uniprot full length protein sequence: <i>221</i></li>\r\n						<li><strong>modification:</strong>name of the modification of the last amino acid according to TopFIND <a href="/modifications">modifications</a>: <i>acteylation</i> or <i>N-acetylaspartate</i></li>\r\n<li><strong>confidence: </strong>a confidence measurement if applicable (e.g. when based on peptide identification by mass spectrometry): <i>0.95</i></li>\r\n						<li><strong>confidence type: </strong>The type of the confidence value given (any of: ''unknown'',''probability'',''MASCOT score'',''X! Tandem score'',''PeptideProphet probability''): <i>probability</i></li>\r\n\r\n						</ul>\r\n					</li>\r\n				</ol>', '', 'Contribute', 1, 1, '2011-04-15 14:52:26', '2011-10-06 19:14:43'),
(40, 'e-intro', '', 'TopFIND requires information on termini, chains, cleavages and interactions to be accompanied by experimental meta-data. This structured information is based on controlled vocabularies where possible and is meant to provide as detailed information as possible. ', '', 'Evidence', 0, 1, '2011-04-15 15:23:59', '2011-06-22 23:11:06'),
(41, 'e_inhibitors', 'Protease inhibitors used', 'Lists the protease inhibitors added during the experiment to ensure that the described effect (terminus, cleavage site, ...) is not affected by proteolytic processes during the sample handling.', 'Lists the protease inhibitors added during the experiment to ensure that the described effect (terminus, cleavage site, ...) is not affected by proteolytic processes during the sample handling.', 'Evidence', 7, 1, '2011-06-22 23:42:16', '2011-06-23 00:21:52'),
(42, 'e_confidence', 'Confidence', 'A measure of the confidence that the identification is correct. The confidence is given as a value between 0 (low) and 1 (high). The type of confidence measure (e.g. MASCOT score or PeptideProphet probability for mass spectrometry based peptide confidence scores) is also given.', 'A measure of the confidence that the identification is correct. The confidence is given as a value between 0 (low) and 1 (high). The type of confidence measure (e.g. MASCOT score or PeptideProphet probability for mass spectrometry based peptide confidence scores) is also given.', 'Evidence', 3, 1, '2011-08-17 17:24:51', '2011-08-17 17:24:51'),
(43, 'e_proteaseassignment', 'Certainty of protease assignment', 'A qualitative measure that the protease assigned to a cleavage is correct. This reflects the possibility of proteolytic activities, other than the one studied and reported, in the experimental system. \r\n<ul>\r\n<li>I â€“ no proteolytic activities present</li>\r\n<li>II â€“proteolytic system present but abolished (e.g. denatured)</li>\r\n<li>III â€“ proteolytic system present but impaired (e.g. inactivated by inhibitors)</li>\r\n<li>IV â€“ proteolytic system present and active (e.g. in vivo study)</li>\r\n<li>unknown</li>', 'A qualitative measure that the protease assigned to a cleavage is correct. This reflects the possibility of proteolytic activities, other than the one studied and reported, in the experimental system. \r\n<ul>\r\n<li>I â€“ no proteolytic activities present</li>\r\n<li>II â€“proteolytic system present but abolished (e.g. denatured)</li>\r\n<li>III â€“ proteolytic system present but impaired (e.g. inactivated by inhibitors)</li>\r\n<li>IV â€“ proteolytic system present and active (e.g. in vivo study)</li>\r\n<li>unknown</li>', 'Evidence', 4, 1, '2011-08-25 23:21:04', '2011-08-26 14:24:58'),
(44, 'e_methodology', 'Method', 'The assay or methodology used is specified in the method field. New or rarely used methods are specified in the Method (other) section. Please contact us if you want to add a widely used method to the list of standard methods.\r\n<ul><li>unknown</li><li>other</li><li>electronic annotation</li><li>COFRADIC</li><li>N-TAILS</li><li>C-TAILS</li><li>ATOMS</li><li>Edman sequencing</li><li>enzymatic biotynilation</li><li>chemical biotynilatin</li><li>MS gel band</li><li>MS semitryptic peptide</li><li>MS other</li><li>mutation analysis</li><ul>', 'The assay or methodology used is specified in the method field. New or rarely used methods are specified in the Method (other) section.', 'Evidence', 3, 1, '2011-10-06 01:15:27', '2011-10-06 01:31:28'),
(45, 'e_method_system', 'Experimental system', 'The experimental system describes the system in which the observation has been made. \r\n<ul><li>unknown</li><li>cell free/li><li>cell culture/li><li>organ or tissue culture/li><li>in vivo sample</li></ul>\r\n', 'The experimental system in which the observation has been made.', 'Evidence', 4, 1, '2011-10-06 01:29:33', '2011-10-06 01:29:33'),
(46, 'e_method_protease_source', 'Protease source', 'For studies investigating the effects of a single protease to discover its substrates, cleavagesites and respective termini the source of the protease (modulation) is stated as ''Protease source'': \r\n<ul><li>unknown</li><li>recombinant</li><li>natural expression</li><li>overexpression</li><li>knock down</li><li>knock out</li></ul>', 'The source or modulation if the effects of a single protease are studied.', 'Evidence', 5, 1, '2011-10-06 01:35:50', '2011-10-06 01:35:50'),
(47, 's_advanced', 'Advanced search', 'The andvance search allows you to specify additional parameter individually or in combination to narrow down your search results. All parameters, including the standard text field can be combined.<br/>\r\nSearch by chromsome location of the encoding genes is limited to <i>Homo sapiens</i> and <i>Mus musculus</i>.<br/>\r\nPlease refere to the search examples and individual help notes for further details.', 'The andvance search allows you to specify additional parameter individually or in combination to narrow down your search results. All parameters, including the standard text field can be combined.<br/>\r\nSearch by chromsome location of the encoding genes is limited to <i>Homo sapiens</i> and <i>Mus musculus</i>.<br/>\r\nPlease refere to the search examples and individual help notes for further details.', 'Search', 5, 1, '2011-10-06 03:25:41', '2011-10-06 21:48:29'),
(48, 's-genomiclocation', 'Search by genomic location', 'To restrict the search to proteins encoded by genes located in a specific chromosome region three search fields are provided in advanced search and filters. A search can be restricted to a single chromosome, a specific chromosome arm or even a single band on a one arm of a chromosome.<br/>\r\nExample: Human p53 chromosome [17] , arm [p] , band [13.1]<br/>\r\nSource: The chromosome location of the encoding gene is derived from ensembl by querying the biomart database with the uniprot accession number.<br/>\r\n[Only available for <i>Homo sapiens</i> and <i>Mus musculus</i>', '(only available for Homo sapiens and Mus musculus)<br/>\r\nYou can restrict the search to proteins encoded by genes located in a specifc genomic region. Please specify at least the chromosome and enter values for arm and band to narrow your search further.', 'Search', 5, 1, '2011-10-06 03:33:48', '2011-10-06 03:49:17'),
(49, 's-arm', 'Chromosome arm', '''p'' or ''q'' to specify the respective chromosome arm.', '''p'' or ''q'' to specify the respective chromosome arm.', 'Search', 7, 1, '2011-10-06 03:35:11', '2011-10-06 03:35:11'),
(50, 's-band', 'Chromosome band', 'The identifier of the band to further narrow down the location on the chromosome (e.g. ''24'' for chromosome 6.p24)', 'The identifier of the band to further narrow down the location on the chromosome (e.g. ''24'' for chromosome 6.p24)', 'Search', 8, 1, '2011-10-06 03:35:51', '2011-10-06 03:35:51'),
(51, 's-numtermini', 'Minimum number of annotated termini', 'The search can be restricted to proteins having annotations for more than any given number of C and/or N termini. To apply this restricting the min number should be entered in the respective fields of the advanced search menu on the front page.', 'Please enter the minimum number of N and/or C termini that need to be annotated for a protein to be listed.', 'Search', 8, 1, '2011-10-06 14:56:09', '2011-10-06 14:56:09'),
(52, 's-chromosomeposition', 'Position on chromosome', 'To narrow down the position of protein encoding genes on the chromosome enter part (e.g. ''p'' for the p arm of a human chromosome) or full (e.g. ''p24'' for band 24 on the p arm of a human chromosome) positional information into the ''Position on chromosome'' field.<br/>\r\n(only supported for <i>Homo sapiens</i> and <i>Mus musculus</i>)', 'Limit by chromosome position:<br/>\r\ne.g. ''p'' or ''p24'' in <i>Homo sapiens</i> or ''B3.1'' for <i>Mus musculus</i><br/>\r\n(only supported for <i>Homo sapiens</i> and <i>Mus musculus</i>)', 'search', 3, 1, '2011-10-06 15:25:38', '2011-10-06 15:42:34'),
(53, 'ts-sequencestart', 'Search for termini by starting sequence', 'Protein N-termini matching a a specific start sequence can be searched for by entering one or several amino acids (single letter notation, upper case) to the sequence start field on the <a href="topfin/ntermini">N termini</a> search page.', 'Enter one or several amino acids (single letter notation, upper case) to search for termini that carry this sequence at their start.', 'Termini', 1, 1, '2011-10-06 16:32:48', '2011-10-06 16:32:48'),
(54, 'ts-sequenceend', '', 'Protein C-termini matching a specific sequence end can be search for by entering one or several amino acids (single letter notation, upper case) to the sequence end field on the <a href="topfin/ctermini">C termini</a> search page.', 'Enter one or several amino acids (single letter notation, upper case) to search for termini that carry this sequence at their end.', 'Termini', 2, 1, '2011-10-06 16:34:24', '2011-10-06 16:34:24'),
(55, 'ts-regexp', 'Search for termini matching a sequence pattern', 'You can search for termini matching a complex sequence pattern or motif by entering a regular expression in the ''Sequence matches regular expression'' field on the <a href="topfin/ntermini">N termini</a> or <a href="topfin/ctermini">C termini</a> search page.<br/>\r\n<b>Regular expression syntax:</b>\r\n<table>\r\n		      <tbody><tr>\r\n		        <td><code>[MRK]</code></td>\r\n		        <td>A single amino acid: M, R or K</td>\r\n		      </tr>\r\n		\r\n		      <tr>\r\n		        <td><code>[^MRK]</code></td>\r\n		        <td>Any single amino acid <em>but</em> M, R, or K</td>\r\n		      </tr>\r\n		      <tr>\r\n		        <td><code>[A-Y]</code></td>\r\n		\r\n		        <td>Any single amino acid in the range a-y (alphabetically)</td>\r\n		      </tr>\r\n		      <tr>\r\n		\r\n		        <td><code>^</code></td>\r\n		        <td>Start of sequence</td>\r\n		      </tr>\r\n		      <tr>\r\n		        <td><code>$</code></td>\r\n		        <td>End of sequence</td>\r\n		      </tr>\r\n		      <tr>\r\n		        <td><code>.</code></td>\r\n		        <td>Any single amino acid</td>\r\n		\r\n		      </tr>\r\n		  <tr>\r\n		    <td><code>(L|I)</code></td>\r\n		    <td>L or I</td>\r\n		  </tr>\r\n		  <tr>\r\n		    <td><code>A?</code></td>\r\n		    <td>Zero or one of A</td>\r\n		\r\n		  </tr>\r\n		  <tr>\r\n		    <td><code>A*</code></td>\r\n		    <td>Zero or more of A</td>\r\n		  </tr>\r\n		  <tr>\r\n		    <td><code>A+</code></td>\r\n		\r\n		    <td>One or more of A</td>\r\n		  </tr>\r\n		  <tr>\r\n		    <td><code>A{3}</code></td>\r\n		    <td>Exactly 3 of A</td>\r\n		  </tr>\r\n		  <tr>\r\n		\r\n		    <td><code>A{3,}</code></td>\r\n		    <td>3 or more of A</td>\r\n		  </tr>\r\n		  <tr>\r\n		    <td><code>A{3,6}</code></td>\r\n		    <td>Between 3 and 6 of A</td>\r\n		  </tr>        \r\n		\r\n		</tbody></table>', 'Enter a regular expression to find termini with matching sequence patterns. See the quick reference box below for options and syntax.', 'Termini', 3, 1, '2011-10-06 17:07:39', '2011-10-06 17:07:39'),
(56, 'ts-pos', 'Search for termini by their position in the genetically encoded protein sequence', 'N termini starting and C termini stopping [before], exactly [at] or [after] a given position (relative to the gene encoded full length protein sequence) can be searched by using the terminus starts and terminus ends search options on the <a href="topfin/ntermini">N termini</a> and <a href="topfin/ctermini">C termini</a> search page respectively.', 'Enter the sequence position relative to the gene encoded full length sequence and select if the termini have to start (N termini) or end (C termini) before, exactly at or after the entered position.', 'Termini', 4, 1, '2011-10-06 17:12:17', '2011-10-06 17:12:17'),
(57, 'e_perturbation', 'Perturbation', 'The type of perturbation used to shift the biological system from the normal physiological condition into a specific state. E.g. chemical induction of apoptosis, gene knock out, over expression of specific genes, ...', 'The type of perturbation used to shift the biological system from the normal physiological condition into a specific state. E.g. chemical induction of apoptosis, gene knock out, over expression of specific genes, ...', 'Evidence', 0, 1, '2013-05-29 19:09:39', '2013-05-29 19:09:39');
